### PR TITLE
Add interfaces for various key handlers

### PR DESCRIPTION
**Pull request by sethvargo**
* Add new interfaces for in-memory key management (used for testing)

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: pkg/keys/in_memory.go [hunk 1]
```diff
@@ -43,42 +49,68 @@ func NewInMemory(ctx context.Context) (*InMemory, error) {
...
+	return key, nil
+}
+
+// AddEncryptionKey stores the key on the system.
+func (k *InMemory) AddEncryptionKey(keyID string, key []byte) error {
+	k.mu.Lock()
+	defer k.mu.Unlock()
 
+	if _, ok := k.cryptoKeys[keyID]; ok {
+		return fmt.Errorf("key already exists: %v", keyID)
```
Comment (chaodaiG) [inline pkg/keys/in_memory.go:110]: %q?

File: pkg/keys/in_memory.go [hunk 2]
```diff
@@ -43,42 +49,68 @@ func NewInMemory(ctx context.Context) (*InMemory, error) {
...
-func (k *InMemory) AddEncryptionKey(keyID string) error {
+// CreateEncryptionKey generates and stores new encryption key identified by the
+// provided keyID.
+func (k *InMemory) CreateEncryptionKey(keyID string) ([]byte, error) {
 	k.mu.Lock()
 	defer k.mu.Unlock()
 
 	if _, ok := k.cryptoKeys[keyID]; ok {
-		return fmt.Errorf("key already exists: %v", keyID)
+		return nil, fmt.Errorf("key already exists: %v", keyID)
```
Comment (chaodaiG) [inline pkg/keys/in_memory.go:92]: %q?

File: pkg/keys/in_memory.go (hunk 1 above)
Comment (sethvargo) [reply pkg/keys/in_memory.go:110]: `%v` is "best for the thing (which is just the string value). `%q` would quote the string, which isn't desirable here. You could argue for `%s`, which is more technically correct, but this is only used for testing anyway, so I'm not super worried 😄

Comment (chaodaiG) [reply pkg/keys/in_memory.go:110]: make sense
/unhold

**Pull request merged**